### ISSUE TITLE
cannot add query parameters to put request

### ISSUE THREAD

**Issue by svrolanski** [hgoebl/DavidWebb#30]
When you use put(), calls to param() are ignored.

**Comment by arbor95** [comment]
you can use 
uri += "?" + WebbUtils.queryString(request.params)

**Comment by svrolanski** [comment]
yes, that is a valid wordaround for it, I think that was what I used, but it is will a bug isn't?

**Comment by arbor95** [comment]
I think you can have a body for put() (same as for post()), where the params go automatically (but I don't know if that is true. For get you surely have no request body, so params go to the queryString.).

**Comment by hgoebl** [comment]
I'm not sure. Having query-params in URL of POST or PUT request can be useful, but AFAIK there are backend stacks which don't support it. At least for `application/x-www-form-urlencoded` it can be ambiguous.

Does anybody know facts about this?

**Comment by svrolanski** [comment]
After some quick reading, it does not seem to break any RFC and I agree with what one said: "The difference between /user/john and /?user=john is merely a semantic one".

https://stackoverflow.com/questions/611906/http-post-with-url-query-parameters-good-idea-or-not

https://stackoverflow.com/questions/14551194/how-are-parameters-sent-in-an-http-post-request/26717908#26717908
